### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by epgui**
These minimal changes remove goreutilities as a dependency and makes everything in the tutorial work with python 3.

### CODE REVIEW

File: FlowCytometryTools/lib/string.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
```
Comment (eyurtsev) [inline FlowCytometryTools/lib/string.py]: `string` is a built-in python package, so we should avoid shadowing it. try using a different name

**Comment by eyurtsev** [comment]
Thanks for taking the time to update this package! Much appreciated! 

I left a few comments on the diff. I don't know if you've used `nose` before, but if you haven't, you can install `nose` and run `nosetests` in the root directory. it'll run through a few of the tests and make troubleshooting a bit faster

I can look later into why travis is failing

**Comment by epgui** [comment]
Hey Eugene,

I'm sorry this is really the first time I do any python, so my commits are likely messy and rather gauche. Thanks for the feedback! I'll keep working on this.

Guillaume

**Comment by epgui** [comment]
Also it seems like travis is running tests for 2.x python... I didn't even know about Travis until now, but I'll look into it on my end as well! :)

**Comment by epgui** [comment]
I think starting with Python 3.3, the __init__.py is no longer required for all packages. [There may be more subtleties involved](https://www.python.org/dev/peps/pep-0420/), but I think these are called namespace packages. I've read somewhere [that best practice with Python 3 was now to do all imports with absolute paths](https://www.python.org/dev/peps/pep-0328/), so if I understand all of that correctly (and please correct me if I'm wrong), lib here will probably never collide with anything.

**Comment by epgui** [comment]
The travis build fails for python 2.7 but now succeeds for python 3.6. I've modified one of the tests, and I feel like that may be bad practice, but I think this has to do with how lists and string evaluations changed from 2.x to 3.x. The problem was that passing a string like "FL-1" as a parameter resulted in a list length of 4 (with index 1 being the character "F") in python 3. Implicit type coercion seems to have changed significantly with 3.x.

I'd like to think python 2.7 support could (and maybe should?) be dropped, but I'd like to hear your thoughts on that before I remove the 2.7 environments from the travis matrix, or before I spend more time trying to get the code to support both 2.7 and 3.6.

**Comment by eyurtsev** [comment]
I hesitate dropping python2.7 support -- there's are still a lot of people that use python2.7 (ubuntu 14.04 LTS is still maintained and was released with python2.7 as the default interpreter), so I'd like to retain backwards compatibility for 1-2 more years. 

`six` is used often to ensure backwards-compatibility 

Here's a reference: http://python-future.org/compatible_idioms.html

Once the code is compatible with python3, we can update travis to test with both interpreters simultaenously

**Comment by xguse** [comment]
People really REALLY need to stop using py27. 

The people in biology who were staying on py27 were mostly bc packages they needed were stuck there. But now, the problem is stuff like this library that think people are all on py27 and don't support py3. Numpy, sklearn, scipy: pretty much **everyone**  supports py3 now and the need to maintain py27 is the big problem. The sooner people abandon it the better.

/lowers-self-from-soapbox

**Comment by eyurtsev** [comment]
It seems like no matter what I do something will get angry :)

Keep in mind, that the concern here is about a small minority of folks that are stuck with legacy systems for a good reason, not an assumption that everyone uses py27. 

Adding backwards compatibility is not very challenging in this case -- the library is very small -- If you're interested in adding six support to this PR, I'll be happy to merge

I'll probably do the full port at some point when I have free time, but this may not be for a few months (I no longer work with flow data)

**Comment by xguse** [comment]
No no. Not angry with you. Just frustrated with scientific software that exists as only py27 in 2018 in general.

Also I don;t think your example of default python installs is a "good reason". You should not be installing into your system's default python environment anyway. Conda and virtualenv has become widespread, solves pretty much if not all of these problems, and is by far the "best practices" behavior. 

In any case.  I promise am not angry and did not mean any bad spirits. 

Just really wish I could quickly and easily incorporate your lib's use with modern code without  having to maintain multiple python interpreter environments for a single task.

**Comment by eyurtsev** [comment]
Merging into a new python3 branch and commandeering

**Pull request merged**